src/js/components/CheckboxTable.js

```diff
@@ -0,0 +1,201 @@
+    let checkedIDs;
+
+    if (isChecked) {
+      checkedIDs = Object.keys(checkedItemsMap);
+      checkedIDs.push(rowID);
+    } else {
+      delete checkedItemsMap[rowID];
+      checkedIDs = Object.keys(checkedItemsMap);
+    }
+
+    onCheckboxChange(checkedIDs);
+  }
+
+  handleHeadingCheckboxChange(prevCheckboxState, eventObject) {
+    let isChecked = eventObject.fieldValue;
+    this.bulkCheck(isChecked);
+  }
+
+  bulkCheck(isChecked) {
+    let checkedIDs = [];
+    let {data, onCheckboxChange} = this.props;
+
+    if (isChecked) {
+      data.forEach(function (datum) {
+        checkedIDs.push(datum.id);
+      });
+      return onCheckboxChange(checkedIDs);
+    }
+
+    onCheckboxChange(checkedIDs);
+  }
+
+  getLabelClass() {
+    return classNames(
+      'form-row-element form-element-checkbox inverse',
+      this.props.labelClass
+    );
+  }
+
+  renderHeadingCheckbox() {
+    let checked = false;
+    let indeterminate = false;
+    let {checkedItemsMap, data} = this.props;
+    let checkedItemsCount = Object.keys(checkedItemsMap).length;
+
+    if (checkedItemsCount > 0) {
+      indeterminate = true;
+    } else {
+      checked = false;
+    }
+
+    if (checkedItemsCount === data.length && checkedItemsCount !== 0) {
+      checked = true;
+    }
+
+    return (
+      <Form
+        formGroupClass="form-group flush-bottom"
+        definition={[
+          {
+            checked,
+            value: checked,
+            fieldType: 'checkbox',
+            indeterminate,
+            labelClass: this.getLabelClass(),
+            name: 'headingCheckbox',
+            showLabel: false
+          }
+        ]}
+        onChange={this.handleHeadingCheckboxChange} />
+    );
+  }
+
+  renderCheckbox(prop, row) {
+    let {checkedItemsMap, uniqueProperty} = this.props;
+    let rowID = row[uniqueProperty];
+    let checked = null;
+
+    if (checkedItemsMap[rowID]) {
+      checked = true;
+    }
+
+    return (
+      <Form
+        formGroupClass="form-group flush-bottom"
+        definition={[{
+          checked,
+          value: checked,
+          fieldType: 'checkbox',
+          labelClass: this.getLabelClass(),
+          name: rowID,
+          showLabel: false
+        }]}
+        onChange={this.handleCheckboxChange} />
+    );
+  }
+
+  getTableRowOptions(row) {
+    let {checkedItemsMap, uniqueProperty} = this.props;
+    if (checkedItemsMap[row[uniqueProperty]]) {
+      return {className: 'selected'};
+    }
+
+    return {};
+  }
+
+  getColumns() {
+    let {getClassName} = ResourceTableUtil;
+
+    return [
+      {
+        className: getClassName,
+        headerClassName: getClassName,
+        prop: 'selected',
+        render: this.renderCheckbox,
+        sortable: false,
+        heading: this.renderHeadingCheckbox
+      }
+    ].concat(this.props.columns);
+  }
+
+  render() {
+    let {className, data} = this.props;
+    let columns = this.getColumns();
+    let sortProp = columns[1].prop;
```

Comment:
maybe set an prop instead of hardcoding column 1? or make a comment? looks kinda mysterious/magical at first glance